mooncake-store/include/allocation_strategy.h

```diff
@@ -24,71 +24,159 @@ class AllocationStrategy {
      * @param allocators Container of mounted allocators, key is segment_name,
      *                  value is the corresponding allocator
      * @param objectSize Size of object to be allocated
+     * @param config Replica configuration
      * @return Selected allocator; returns nullptr if allocation is not possible
      *         or no suitable allocator is found
      */
     virtual std::unique_ptr<AllocatedBuffer> Allocate(
         const std::unordered_map<std::string, std::shared_ptr<BufferAllocator>>&
             allocators,
-        size_t objectSize) = 0;
+        size_t objectSize, const ReplicateConfig& config) = 0;
 };
 
+/**
+ * @brief Random allocation strategy with local preference support.
+ *
+ * This strategy first attempts to allocate from a preferred segment if
+ * specified, then falls back to random allocation among all available
+ * allocators.
+ */
 class RandomAllocationStrategy : public AllocationStrategy {
```

Comment:
How about a new Class? The default allocation strategy is random, we can choose local preference allocation strategy in run time command lines or environment variables

Reply:
If the client leaves `preferred_segment` unset in the replica config, it'll just go back to the old random way. But if they do set it, it's definitely supposed to stick to what the config says.

So, I don't think we need a whole new class for this.

I only messed with this config in the Python bindings so far. The C++ side should still be doing its thing like before.

Reply:
Sounds fine, another question, when a client has a huge put to its preferred segments, how to re-balance the memory usage? By gc or leave it to user control?

Reply:
We're planning to add support for segment-level metrics